modepy/quadrature/jacobi_gauss.py

```diff
@@ -261,9 +290,94 @@ def legendre_gauss_lobatto_nodes(
         backend: Optional[str] = None,
         force_dim_axis: bool = False) -> np.ndarray:
     """Compute the Legendre-Gauss-Lobatto quadrature nodes.
-    *N* is the number of points.
+    *N+1* is the number of nodes.
 
     Exact to degree :math:`2N - 1`.
     """
     return jacobi_gauss_lobatto_nodes(0, 0, N,
             backend=backend, force_dim_axis=force_dim_axis)
+
+
+class JacobiGaussLobattoQuadrature(Quadrature):
+    r"""
+    Compute the Jacobi-Gauss-Lobatto quadrature with respect
+    to the Jacobi weight function
+
+    .. math::
+
+        I[f] = \int_{-1}^1 f(x) (1 - x)^\alpha (1 + x)^\beta\, \mathrm{d}x,
+
+    There will be *N+1* nodes. Exact to degree :math:`2N - 3`.
+
+    .. versionadded:: 2024.2
+    """
+
+    def __init__(self,
+            alpha: float, beta: float, N: int,  # noqa: N803
+            *, backend: Optional[str] = None,
+            force_dim_axis: bool = False) -> None:
+        nodes = jacobi_gauss_lobatto_nodes(alpha, beta, N, backend)
+
+        from math import gamma
+
+        from modepy.modes import binom, scaled_jacobi as jacobi
+
+        # Formula numbers refer to https://doi.org/10.1023/A:1016689830453
+        if N + 1 < 2:
+            raise ValueError("Lobatto rules must have at least two nodes")
+
+        n = N - 1
+
+        # FIXME: Recurrences might be better than just typing up the formulas.
+
+        # (3.10)
+        common_factor = (
+            2**(alpha + beta + 1)
+            * binom(n + alpha + 1, n)
+            / binom(n + beta + 1, n)
+            / binom(n + alpha + beta + 2, n)
+        )
+        edge_weight = (
+            common_factor
+            * gamma(alpha + 2) * gamma(beta + 1)
+            / gamma(alpha + beta + 3)
+        )
+
+        # (4.7)
+        int_nodes = nodes[1:-1]
+        frac = (
+            4*(n+alpha+1) * (n+beta+1) + (alpha-beta)**2
+        ) / (
+            2*n + alpha + beta + 2
+        )**2
+
+        int_weights = (
+            common_factor
+            * gamma(alpha + 2) * gamma(beta + 2)
+            / gamma(alpha + beta + 3)
+
+            # FIXME: This is part of the paper, but tests only pass without it:
+            # / (n + 1)**2
```

Comment:
Hm, it seems like that should be there from (4.2), but couldn't spot what could be wrong :\